### DISCUSSION TITLE
The theme language affect search result?

### DISCUSSION THREAD

**Discussion by halx99** [squidfunk/mkdocs-material#2191] [Q&A] [answered]
Hello,  I have question about search zh with en.

search words: set_keepalive

There are to revision docs urls:
1. https://docs.yasio.org/zh_CN/0.0.1/
Can't search chinese mix with english

2. https://docs.yasio.org/zh_CN/latest/
Can search chinese mix with english

the different with above two doc sites config is theme language: 0.0.1 is 'zh', latest is 'en'.

**Comment by squidfunk** [comment]
Search works for the second link you posted!? I can search in English and Chinese.

**Comment by halx99** [comment]
Yes, because I set theme lang to 'en' for second link: https://github.com/yasio/docs-936/blob/master/mkdocs.yml#L9

And the first link theme lang is 'zh': https://github.com/yasio/docs-936/blob/0.0.1/mkdocs.yml#L9

**Reply by squidfunk** [reply]
Could you try switching `ja` and `en` in the search plugin configuration and see if that resolves the issue:

``` yaml
plugins:
  - search:
      lang:
        - ja
        - en
```

**Comment by halx99** [comment]
My mkdocs.yml file already have follow config: 
```yml
plugins:
  - search:
      lang:
        - ja
        - en
```

Please see:
https://github.com/yasio/docs-936/blob/0.0.1/mkdocs.yml#L56

But still doesn't work.

**Reply by squidfunk** [reply]
I see. Could you please provide a minimum reproducible test case and file an issue? Please follow the issue template and make sure that the error is reproducible, i.e. provide a test document with English and Chinese + your `mkdocs.yml` etc.

**Reply by halx99** [reply]
Thanks for your reply, I will file a issue later.

**Reply by halx99** [reply]
Hi, squidfunk
  The minimum reproducible test case is here: https://github.com/halx99/search-test
And I have test other themes listed on mkdocs github wiki page, no one support `zh` search even through I set the theme language to `en`

**Reply by squidfunk** [reply]
Thanks for providing the minimal reproduction. Indeed, it doesn't work properly when setting the language to `zh`. However, if I remove the search plugin configuration, I can search for English and Chinese:

https://user-images.githubusercontent.com/932156/104091572-7b203f80-527e-11eb-8027-613e5e48d70a.mp4

Please note that Chinese search support is still experimental, as [we're using the Japanese language stemmer for Chinese](https://github.com/squidfunk/mkdocs-material/blob/5c36dfbbfd1412d1c71e855fff56990d01e26f4c/src/partials/languages/zh.html#L33). [`lunr-languages`](https://github.com/MihaiValentin/lunr-languages) doesn't offer a dedicated stemmer for Chinese, but Japanese works reasonably well. Thus, we cannot improve Chinese support on our end - it's a matter of `lunr-languages`.

**Reply by halx99** [reply]
Thanks for your answer, I got it.
I like mkdocs-material

**Reply by halx99** [reply]
And another Note:

I view your movie, when remove plugin search configuration, and theme lang is `zh`, the `en` search not works properly:

1. no highlight
2. search words will break with charactor `_`

**Reply by squidfunk** [reply]
1. Yes, it's not perfect and probably never will be. Material for MkDocs supports more than 40 languages, it's very hard to make sure everything works perfectly.
2. Please see [configuring the search tokenizer](https://squidfunk.github.io/mkdocs-material/setup/setting-up-site-search/#separator) - it may solve both issues.

**Reply by halx99** [reply]
Thanks, I configure the search tokenizer, and the highlight issue solved, but words still break with `'_'`:
```yml
plugins:
  - search:
      separator: '[\s\-\.]+'
```

**Reply by halx99** [reply]
When I change the file https://github.com/squidfunk/mkdocs-material/blob/5c36dfbbfd1412d1c71e855fff56990d01e26f4c/src/partials/languages/zh.html#L34 
search.config.pipeline as same with `en.html`, then the mix search work properly.
So could we provide a configuration value `search.pipeline` for user, just like `search.separator`?

**Reply by squidfunk** [reply]
You can add the underscore to the regex:

``` yaml
plugins:
  - search:
      separator: '[\s\-\._]+'
```

**Reply by halx99** [reply]
Actually, I don't want word stop with character `_`, it's useful for API documentation, such as c++ opensource library may contains many APIs with character `_`

**Comment by squidfunk** [comment, marked as answer]
Chinese search support is still experimental, as [we're using the Japanese language stemmer for Chinese](https://github.com/squidfunk/mkdocs-material/blob/5c36dfbbfd1412d1c71e855fff56990d01e26f4c/src/partials/languages/zh.html#L33). [`lunr-languages`](https://github.com/MihaiValentin/lunr-languages) doesn't offer a dedicated stemmer for Chinese, but Japanese works reasonably well. The following configuration was reported to work for both Chinese and English:

``` yaml
theme:
  language: zh
plugins:
  - search
```
